app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,62 @@
+class PasswordResetsController < ApplicationController
+  def new
+    if logged_in?
+      redirect_to root_url
+      flash[:message] = "You are already logged in."
+    end
+  end
+
+  def create
+    @user = User.find_by(email: params[:password_reset][:email])
+    if @user.present?
+      token = SecureRandom.urlsafe_base64
+      @user.update_attributes({ reset_token: token,
+                                reset_digest: digest(token) })
+      UserMailer.password_reset(@user).deliver_now
+      redirect_to root_url
+      flash[:message] = "Please check your email to reset your password."
+    else
+      redirect_to new_password_reset_path
+      flash[:danger] = "Please provide a valid email address."
+    end
+  end
+
+  def edit
+    @user = User.find_by(email: params[:email])
+    if !(@user.present? && @user.authenticated?(:reset, params[:id]))
+      redirect_to root_url
+      flash[:danger] = "Not a valid password reset request."
+    end
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if valid_params?
+      if @user.update_attributes(user_params)
```

Comment:
You really want these to be on one line:
```ruby
if valid_params? && @user.update_attributes(user_params)
```
The way you have it now, it's possible that `valid_params?` passes, but `update_attributes` fails, in that case, this method would return `nil` and Rails would try to render the `update` template. Which would be bad.